src/scipp/ndimage/__init__.py

```diff
@@ -0,0 +1,247 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+# @author Simon Heybrock
+"""Sub-package for multidimensional image processing.
+
+This subpackage provides wrappers for a subset of functions from
+:py:mod:`scipy.ndimage`.
+"""
+from functools import wraps
+from typing import Callable, Dict, Union
+
+import scipy.ndimage
+
+from ..core import Variable, DataArray
+from ..core import CoordError, DimensionError, VariancesError
+from ..core import empty_like, islinspace, ones
+
+
+def _ndfilter(func: Callable) -> Callable:
+
+    @wraps(func)
+    def function(x: Union[Variable, DataArray], **kwargs) -> Union[Variable, DataArray]:
+        if 'output' in kwargs:
+            raise TypeError("The 'output' argument is not supported")
+        if x.variances is not None:
+            raise VariancesError(
+                "Filter cannot be applied to input array with variances.")
+        if getattr(x, 'masks', None):
+            raise ValueError("Filter cannot be applied to input array with masks.")
+        return func(x, **kwargs)
+
+    return function
+
+
+def _delta_to_positional(x: Union[Variable, DataArray], dim, index, dtype):
+    if not isinstance(index, Variable):
+        return index
+    coord = x.coords[dim]
+    if not islinspace(coord, dim).value:
+        raise CoordError(
+            f"Data points not regularly spaced along {dim}. To ignore this, "
+            f"provide a plain value (convertible to {dtype.__name__}) instead of a "
+            "scalar variable. Note that this will correspond to plain positional "
+            "indices/offsets.")
+    pos = (len(coord) - 1) * (index.to(unit=coord.unit) / (coord[-1] - coord[0])).value
+    return dtype(pos)
+
+
+def _positional_index(x: Union[Variable, DataArray], index, name=None, dtype=int):
+    if not isinstance(index, dict):
+        return [_delta_to_positional(x, dim, index, dtype=dtype) for dim in x.dims]
+    if set(index) != set(x.dims):
+        raise ValueError(f"Data has dims={x.dims} but input argument '{name}' provides "
+                         f"values for {tuple(index)}")
+    return [_delta_to_positional(x, dim, index[dim], dtype=dtype) for dim in x.dims]
+
+
+@_ndfilter
+def gaussian_filter(x: Union[Variable, DataArray],
+                    /,
+                    *,
+                    sigma,
+                    order=0,
+                    **kwargs) -> Union[Variable, DataArray]:
+    """
+    Multidimensional Gaussian filter.
+
+    This is a wrapper around :py:func:`scipy.ndimage.gaussian_filter`. See there for
+    full argument description. There are two key differences:
+
+    - This wrapper uses explicit dimension labels in the ``sigma`` and ``order``
+      arguments. For example, instead of ``sigma=[4, 6]`` use
+      ``sigma={{'time':4, 'space':6}}``
```

Comment:
Why double braces?

Reply:
This is an fstring, double to escape them.

Reply:
This one isn't. Only the one in `footprint_filter` is.

Reply:
🤦